### PR TITLE
Added constants to the FHIR Path tool

### PR DESCRIPTION
**Pull request by carlosfelix2**
This PR adds the ability to add constants to the FHIR Path tool.  For example:
```
./prime fhirpath -i src/testIntegration/resources/datatests/HL7_to_FHIR/sample_co_1_2022051 -c patientState=Bundle.entry.resource.ofType(Patient).address.state -c patientZip=Bundle.entry.resource.ofType(Patient).address.postalCode -c patientCountry=Bundle.entry.resource.ofType(Patient).address.country
```

Test Steps:
1. Run `./gradlew fatjar`
2. Run the prime command above in the description.
3. Type in `%patientZip` and press Enter. Verify the zip code value is provided.

## Changes
- Added constants to the FHIR path tool

### Testing
- [x] Tested locally?
- [ ] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [ ] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?

### CODE REVIEW

**Comment by victor-chaparro** [comment]
I couldn't find any unit tests on the CLI commands. Are they not required, are these mostly used for integration testing?

**Comment by carlosfelix2** [comment]
> I couldn't find any unit tests on the CLI commands. Are they not required, are these mostly used for integration testing?

Yes, the CLI is a test tool, so not unit tests required for a test tool

**Review by jimmyfagan — APPROVED** [review]
A few comments related to the description:

- Had a slight issue with the command example used to test this, the `-i` value didn't point to a file in the repo, maybe should edit that.
- On Mac the const paths had to be in quotes, if that works on Windows too, it might be worth updating
- Is this related to an Issue number? Didn't see it linked in the description, wasn't sure if that was by mistake or not.

The code looks good, marking approved, thanks!

**Review by TurkeyFried — APPROVED** [review]
Was able to test, warnings appeared when expected. Very useful change!

File: prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/ConstantResolver.kt [hunk 1]
```diff
@@ -329,7 +330,9 @@ class FhirPathCustomResolver : FHIRPathEngine.IEvaluationContext, Logging {
         return if (constantValue.isNullOrBlank()) null
         else {
             val values = FhirPathUtils.evaluate(appContext, appContext.focusResource, appContext.bundle, constantValue)
-            if (values.size != 1) {
+            if (values.isEmpty()) {
+                null
+            } else if (values.size != 1) {
```
Comment (clediggins-usds) [inline prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/ConstantResolver.kt:335]: Concerned at all that if your mistype your constant, or it is multi valued the cli tool just boots you out?

**Review by clediggins-usds — APPROVED** [review]

File: prime-router/src/main/kotlin/cli/ProcessFhirCommands.kt [hunk 2]
```diff
@@ -224,9 +233,15 @@ class FhirPathCommand : CliktCommand(
         if (contents.isBlank()) throw CliktError("File ${inputFile.absolutePath} is empty.")
         val bundle = FhirTranscoder.decode(contents)
         focusResource = bundle
-        fhirPathContext = CustomContext(
-            bundle, bundle, mutableMapOf("rsext" to "'https://reportstream.cdc.gov/fhir/StructureDefinition/'")
-        )
+        val constantList = mutableMapOf("rsext" to "'https://reportstream.cdc.gov/fhir/StructureDefinition/'")
+        constants.entries.forEach {
```
[the remark this answers is not in the document]
Comment (carlosfelix2) [reply prime-router/src/main/kotlin/cli/ProcessFhirCommands.kt:237]: That gets checked by the FHIR Path library at evaluation time.  Same thing in the rest of the code.  I tried to verify this before hand when I implemented the FHIR to HL7 library, but it turns out it is not possible until one has all the data and constant definitions ready = evaluation time.

File: prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/ConstantResolver.kt (hunk 1 above)
Comment (carlosfelix2) [reply prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/ConstantResolver.kt:335]: Added an extra check, so now it will not blow up

**Pull request merged**